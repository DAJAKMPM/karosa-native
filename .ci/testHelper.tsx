import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
