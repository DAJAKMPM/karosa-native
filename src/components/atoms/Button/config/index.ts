/**
 *
 * @format
 *
 */

export const defaultOpacity = 0.8;
