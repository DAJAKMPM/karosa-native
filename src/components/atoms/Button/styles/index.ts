/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { DIMENS, theme } from "@app/styles";

const ButtonStyles = StyleSheet.create({
  container: {
    backgroundColor: theme.colors.primary,
    height: DIMENS.screenHeight * 0.075,
  },
});

export default ButtonStyles;
