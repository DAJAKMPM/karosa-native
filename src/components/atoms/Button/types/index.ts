/**
 *
 * @format
 *
 */

import { StyleProp, TextStyle, ViewStyle } from "react-native";
import { IconNode } from "react-native-elements";

type ButtonType = "solid" | "clear" | "outline";
type Icon = {
  left?: React.ReactElement | undefined;
  right?: React.ReactElement | undefined;
};

export type PropsType = {
  title: string;
  onPress?: () => void;
  type?: ButtonType;
  loading?: boolean;
  disabled?: boolean;
  titleStyle?: StyleProp<TextStyle>;
  buttonStyle?: StyleProp<ViewStyle>;
  icon?: IconNode;

  //from component/button type
  containerStyle?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  iconVariationTwo?: Icon;

  // new
  variation?: number;
};
