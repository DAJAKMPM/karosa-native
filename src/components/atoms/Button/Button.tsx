/**
 *
 * Button
 * @format
 *
 */

import React, { FC } from "react";
import { Button as RnButton } from "react-native-elements";

import type { PropsType } from "./types";
import { OPACITY } from "./config";
import ButtonStyles from "./styles";

const Button: FC<PropsType> = (props) => {
  const {
    title,
    type = "solid",
    disabled,
    loading,
    icon,
    titleStyle,
    buttonStyle,
    onPress,
  } = props;

  return (
    <RnButton
      type={type}
      activeOpacity={OPACITY}
      raised
      icon={icon}
      title={title}
      onPress={onPress}
      loading={loading}
      disabled={disabled}
      titleStyle={titleStyle}
      buttonStyle={[ButtonStyles.container, buttonStyle]}
    />
  );
};

export default Button;
