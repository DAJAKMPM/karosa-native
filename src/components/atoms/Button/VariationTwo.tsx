/**
 *
 * Button
 * @format
 *
 */
import React from "react";
import { Text, TouchableOpacity } from "react-native";

import { PropsType as Props } from "./types";
import { stylesVarTwo as styles } from "./styles";

const VariationTwo: React.FC<Props> = ({
  title,
  onPress,
  iconVariationTwo,
  containerStyle,
  textStyle,
  disabled,
}) => {
  return (
    <TouchableOpacity
      style={[styles.button, containerStyle]}
      onPress={onPress}
      disabled={disabled}>
      {iconVariationTwo?.left}
      <Text style={[styles.text, textStyle]}>{title}</Text>
      {iconVariationTwo?.right}
    </TouchableOpacity>
  );
};

export default VariationTwo;
