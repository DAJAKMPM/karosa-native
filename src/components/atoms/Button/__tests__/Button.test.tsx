import React from "react";
import { noop } from "lodash";
import { renderWithWrapper } from "@app/ci/testHelper";
import Button from "../Button";

describe("Button Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <Button onPress={noop} title="test" />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
