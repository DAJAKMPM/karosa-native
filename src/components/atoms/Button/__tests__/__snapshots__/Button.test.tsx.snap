// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        Object {
          "backgroundColor": "#fff",
          "overflow": "visible",
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#0AA351",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "padding": 16,
            "position": "relative",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          test
        </Text>
      </View>
    </View>
  </View>
</View>
`;
